### PR TITLE
create animal class

### PR DESCRIPTION
**Pull request by codrex**
###  ​What​ ​does​ ​this​ ​PR​ ​​do​? 
create a class of Animal

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default class Animal{
```
Comment (andela-nrotimi) [inline src/main.js]: A space should be applied between the class name and the curly brace

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  constructor(name,legs){
+    const data = {name,legs};
+    this.getName = ()=> data.name;
+    this.getLegs = ()=> data.legs;
+    this.setName = name=>data.name = name;
```
Comment (andela-nrotimi) [inline src/main.js]: Inadequate spacing

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    this.sound = 'Animal make sound';
+  }
+  makeSound(){
+    return `${this.sound}`;
+  }
+}
```
Comment (andela-nrotimi) [inline src/main.js]: No new line added to end of file

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export default class Animal{
+  constructor(name,legs){
+    const data = {name,legs};
+    this.getName = ()=> data.name;
+    this.getLegs = ()=> data.legs;
+    this.setName = name=>data.name = name;
+    this.sound = 'Animal make sound';
+  }
+  makeSound(){
+    return `${this.sound}`;
```
Comment (andela-nrotimi) [inline src/main.js]: why do you have to use a template literal?

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export default class Animal{
+  constructor(name,legs){
+    const data = {name,legs};
+    this.getName = ()=> data.name;
```
Comment (andela-nrotimi) [inline src/main.js]: what are you trying to achieve here?

File: test/test.test.js [hunk 6]
```diff
@@ -1,4 +1,21 @@
...
+    expect(animal.getLegs()).to.equal('4');
+  })
+  it('Encapsulation:: should return undefined',()=>{
+    expect(animal.name).to.equal(undefined);
+  })
+  it(' should return Animal make sound',()=>{
+    expect(animal.makeSound()).to.equal('Animal make sound');
+  })
+
+})
```
Comment (andela-nrotimi) [inline test/test.test.js]: No new line

File: test/test.test.js [hunk 7]
```diff
@@ -1,4 +1,21 @@
 import { expect } from 'chai';
-import {Cat} from '../src/main';
-import {Human} from '../src/main';
+import Animal from '../src/main';
 
+
+
+describe('Testing Animal class function',()=>{
+  const animal = new Animal('Cat','4');
```
Comment (andela-nrotimi) [inline test/test.test.js]: why do you have to use const?

File: src/main.js (hunk 1 above)
Comment (codrex) [reply src/main.js]: ok... will setup eslint to help do the checking next time

File: src/main.js (hunk 5 above)
Comment (codrex) [reply src/main.js]: encapsulation, 
making sure that name and legs  can't be accessed from outside the class

File: src/main.js (hunk 2 above)
Comment (codrex) [reply src/main.js]: ok

File: src/main.js (hunk 4 above)
Comment (codrex) [reply src/main.js]: oh....!! made changes to the function and I forgot it there

File: src/main.js (hunk 3 above)
Comment (codrex) [reply src/main.js]: ok

File: test/test.test.js (hunk 6 above)
Comment (codrex) [reply test/test.test.js]: ok

File: test/test.test.js (hunk 7 above)
Comment (codrex) [reply test/test.test.js]: because am only assigning to the variable once

**Pull request merged**